kv/util/util.go

```diff
@@ -0,0 +1,285 @@
+package util
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/m3db/m3cluster/generated/proto/commonpb"
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3x/log"
+)
+
+var errNilStore = errors.New("kv store is nil")
+
+type getValueFn func(kv.Value) (interface{}, error)
+type updateFn func(interface{})
+
+// WatchAndUpdateBool sets up a watch for a bool property.
+func WatchAndUpdateBool(store kv.Store, name string, property *bool, lock sync.Locker,
```

Comment:
I wonder if this is a good candidate to explore a template based approach to generate code blocks that only differ by their types.